ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdatedV2.java

```diff
@@ -17,17 +17,21 @@
 import org.hyperledger.besu.consensus.merge.blockcreation.MergeMiningCoordinator;
 import org.hyperledger.besu.ethereum.ProtocolContext;
 import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.mainnet.TimestampSchedule;
 
 import io.vertx.core.Vertx;
 
+// TODO Withdrawals use composition instead? Want to make it more obvious that there is no
```

Comment:
Yes, I actually added that for this PR. Forgot to call it out myself, but thought it might be a discussion point for the review.

Currently thinking we inheritance may lead people in the future to add V2+ specific changes in the subclass which I'm currently trying to avoid in order to keep the V1/V2 differences as small as possible. Instead, favour making the code backwards compatible.

Reply:
If we decide on composition over inheritance then I'd like to do that in a separate PR as it will affect three engine API methods, assuming we want them to follow a consistent approach.

Reply:
Think it's ok using inheritance for now